core/src/test/java/org/infinispan/eviction/impl/ExceptionEvictionTest.java

```diff
@@ -481,13 +481,10 @@ public void testEntryExpiration(boolean maxIdle, boolean readInTx) throws Except
          assertNull(cache(0).get(expiringKey));
       }
 
-      // Off heap doesn't expire entries on access yet ISPN-8380
-      // Also max idle in a transaction don't remove entries until reaper runs
-      if (storageType == StorageType.OFF_HEAP || maxIdle) {
-         for (Cache cache : caches()) {
-            ExpirationManager em = cache.getAdvancedCache().getExpirationManager();
-            em.processExpiration();
-         }
+      // Manually triggering batch expiration is no longer required, but it helps reproduce ISPN-12971
```

Comment:
@wburns should we maybe add a test that more forces the parallel expiration of the same key (via get and/or via processExpiration) more reliably?

Reply:
Seems like a good idea to me. 👍

Reply:
I thought it would be a good idea, too, but I'm not so sure how to do it, so I wouldn't hold up this PR.